runtime/src/validate_block/tests.rs

```diff
@@ -152,4 +152,25 @@ fn validate_block_with_extrinsics() {
 		witness_data_storage_root
 	);
 	call_validate_block(client.genesis_hash(), block_data).expect("Calls `validate_block`");
-}
\ No newline at end of file
+}
+
+#[test]
+#[should_panic]
```

Comment:
if it panics internally, you can use `#[should_panic(expected = "error_substring")]`

Reply:
Panic is intercepted internally, so `#[should_panic(expected = "Invalid parent hash")]` will not work.

Reply:
Yep, the panic is thrown in WASM and this just prints it.